drivers/adc/adc_stm32.c

```diff
@@ -811,6 +812,11 @@ static void adc_stm32_setup_channels(const struct device *dev, uint8_t channel_i
 		__LL_ADC_CHANNEL_TO_DECIMAL_NB(ADC_CHANNEL_VREFINT) == channel_id) {
 		adc_stm32_set_common_path(dev, LL_ADC_PATH_INTERNAL_VREFINT);
 	}
+
+	if (config->has_vbat_channel &&
+		__LL_ADC_CHANNEL_TO_DECIMAL_NB(ADC_CHANNEL_VBAT) == channel_id) {
```

Comment:
U5 also has LL_ADC_CHANNEL_VBAT and LL_ADC_CHANNEL_VBAT_ADC4 which looks like a different channel.

Reply:
removing this part from that PR